narwhal/node/src/main.rs

```diff
@@ -206,6 +206,14 @@ async fn run(
     worker_keypair: NetworkKeyPair,
     registry: Registry,
 ) -> Result<(), eyre::Report> {
+    // Only enabled if failpoints feature flag is set
+    if fail::has_failpoints() {
+        warn!("Failpoints are enabled");
+        mem::forget(fail::FailScenario::setup());
```

Comment:
why is this being forgotten?

Reply:
I would expect this to be held in `main` with a guard that is dropped once `main` exits instead of forgetting it.

Reply:
Yup you are right!